MerlinAU.sh

```diff
@@ -4283,9 +4373,306 @@ _EntwareServicesHandler_()
+                BM_VERSION="$(echo "$BM_VERSION" | sed 's/\.\([0-9]\)\([0-9]\)/.\1.\2/')"
+            fi
+
+            # Convert version strings to comparable numbers
+            local currentBM_version="$(_ScriptVersionStrToNum_ "$BM_VERSION")"
+            local requiredBM_version="$(_ScriptVersionStrToNum_ "1.5.3")"
+
+            # Check if BACKUPMON version is greater than or equal to 1.5.3
+            if [ "$currentBM_version" -ge "$requiredBM_version" ]; then
+                # Execute the backup script if it exists #
+                echo ""
+                Say "Backup Started (by BACKUPMON)"
+                sh /jffs/scripts/backupmon.sh -backup >/dev/null
+                BE=$?
+                Say "Backup Finished"
+                echo ""
+                if [ $BE -eq 0 ]; then
+                    Say "Backup Completed Successfully"
+                    echo ""
+                else
+                    Say "Backup Failed"
+                    echo ""
+                    _SendEMailNotification_ NEW_BM_BACKUP_FAILED
+                    _DoCleanUp_ 1
+                    if "$isInteractive"
+                    then
+                        printf "\n${REDct}**IMPORTANT NOTICE**:${NOct}\n"
+                        printf "The firmware flash has been ${REDct}CANCELLED${NOct} due to a failed backup from BACKUPMON.\n"
+                        printf "Please fix the BACKUPMON configuration, or consider uninstalling it to proceed flash.\n"
+                        printf "Resolving the BACKUPMON configuration is HIGHLY recommended for safety of the upgrade.\n"
+                        _WaitForEnterKey_ "$mainMenuReturnPromptStr"
+                        return 1
+                    else
+                        _DoExit_ 1
+                    fi
+                fi
+            else
+                # BACKUPMON version is not sufficient
+                echo ""
+                Say "${REDct}**IMPORTANT NOTICE**:${NOct}"
+                echo ""
+                Say "Backup script (BACKUPMON) is installed; but version $BM_VERSION does not meet the minimum required version of 1.5.3."
+                Say "Skipping backup. Please update your version of BACKUPMON."
+                echo ""
+            fi
+        else
+            Say "Backup script (BACKUPMON) is disabled in the advanced options. Skipping backup."
+            echo ""
+        fi
+    else
+        Say "Backup script (BACKUPMON) is not installed. Skipping backup."
+        echo ""
+    fi
+    return 0
+}
+
+##---------------------------------------##
+## Added by ExtremeFiretop [2024-Jul-23] ##
+##---------------------------------------##
+_RunManualUpdateNow_()
+{
+    OfflineConfigFile="${SETTINGS_DIR}/offline_updates.txt"
+    if [ ! -f "$OfflineConfigFile" ]; then
+        _DoExit_ 1
+    fi
+    # Source the configuration file
+    . "$OfflineConfigFile"
+
+    # Check required parameters
+    if [ "$FW_OFFLINE_UPDATE_IS_ALLOWED" != "true" ]; then
```

Comment:
This conditional statement resulted in a "parameter not found" error during testing if the offline config file existed but was empty, or it had the wrong parameter name (e.g. misspelled variable), or the parameter line was commented out.

Reply:
Another great catch!
Totally didn't think of testing a misspelled variable, wrong parameter name, etc.